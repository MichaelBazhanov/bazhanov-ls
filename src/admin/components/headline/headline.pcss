.headline-component {
	background-color: $bg-head-color;

}
.container-headline {
	display: flex;
	align-items: center;
	padding: 15px 0;
}
.title {
	color: rgba($text-color, .5);
	font-size: 14px;
	line-height: 19px;
	margin-right: auto;
	margin-left: 25px;
}
.btns {
	margin-left: auto;
	display: flex;
	justify-content: center;
	align-items: center;
}
.btn {
	font-size: 16px;
	line-height: 22px;
	text-decoration: underline;
	color: rgba($text-color, .9);
	cursor: pointer;
	background-color: transparent;
	border: none;
}