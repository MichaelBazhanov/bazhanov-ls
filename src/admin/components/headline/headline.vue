<template>
	<div class="headline-component">
		<div class="container">
			<slot />
			<div class="title">{{title}}</div>
			<div class="btns">
				<button type="button" class="btn" @click="$emit('change', 'message')">Сообщения</button>
				<button type="button" class="btn" @click="$emit('change', 'logout')">Выйти</button>
			</div>
		</div>
	</div>
</template>

<style lang="postcss" scoped src="./headline.pcss"></style>

<script>
export default {
	props: {
		title: String,
	}
}
</script>

