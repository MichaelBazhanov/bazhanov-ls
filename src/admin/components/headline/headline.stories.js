import headline from "./headline.vue";
import { action } from "@storybook/addon-actions";

const methods = {
	onChange: action('onChange_up') //onChange2 всплывает с данными из реального Vue компонента
}

export default {
	title: 'headline',
	components: { headline }
}

export const defaultView = () => ({
	template: `
		<headline
			@change="onChange"
			title="Панель управления"
		>
			<h2>это тестовый текст</h2>
		</headline>
	`,
	components: { headline },
	methods,
})

defaultView.story = {
	name: 'Стандартный вид',
}