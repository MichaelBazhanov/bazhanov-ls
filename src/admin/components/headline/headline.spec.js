import headline from './headline'
import { shallowMount } from "@vue/test-utils"

it('headline snapshot + emitted()', async () => {
	//steps
	const wrapper = shallowMount(headline)

	//actions
	await wrapper.find('.btn').trigger('click');

	//check
	await wrapper.vm.$nextTick(); //ожидаем асинхронного обновления всего
	expect(wrapper.emitted('logout')[0]).toEqual(['logout'])
	expect(wrapper.element).toMatchSnapshot()
})