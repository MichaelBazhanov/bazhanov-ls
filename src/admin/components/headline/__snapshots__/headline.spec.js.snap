// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headline snapshot + emitted() 1`] = `
<div
  class="headline-component"
>
  <div
    class="container container-headline"
  >
     
    <div
      class="title"
    >
      
    </div>
     
    <div
      class="btns"
    >
      <button
        class="btn"
        type="button"
      >
        Выйти
      </button>
    </div>
  </div>
</div>
`;
